src/drift/Arguments/ArgumentOptions.cs

```diff
@@ -0,0 +1,75 @@
+﻿using CommandLine;
+using System;
+using System.IO;
+
+namespace rangers.antidrift.drift.Arguments
+{
+    public class ArgumentOptions
+    {
+        private string configurationFilePath;
+        private string serviceUrl;
+
+        [Option('c', "config-file",
+            Required = true,
+            HelpText = "Path to the configuration file.")]
+        public string ConfigurationFilePath
+        {
+            get
+            {
+                return configurationFilePath;
+            }
+
+            set
+            {
+                string fullpath = Path.GetFullPath(value);
+
+                if (!File.Exists(fullpath))
+                {
+                    throw new FileNotFoundException($"Configuration file was not found at '{fullpath}' ", fullpath);
+                }
+
+                configurationFilePath = fullpath;
+            }
+        }
+
+        [Option('s', "service-url",
+            Required = true,
+            HelpText = "URL to the service you will connect to, e.g. https://youraccount.visualstudio.com/DefaultCollection")]
+        public string ServiceUrl
+        {
+            get { return serviceUrl; }
+            set
+            {
+                if (!Uri.IsWellFormedUriString(value, UriKind.Absolute))
+                {
+                    throw new Exception($"URL '{value}' is not a valid URL.");
+                }
+
+                serviceUrl = value;
+            }
+        }
+
+        [Option('a', "auth-type",
+            Default = AuthType.Pat,
+            HelpText = "Method of authentication")]
+        public AuthType AuthType { get; set; }
+
+        [Option('t', "token",
+            Required = true,
```

Comment:
Should this and the User/Password be marked as Required? Seems like that would conflict, unless the SetName is splitting them up

Reply:
They should as the SetName is able to split them up, mandatory only if he identifies that a certain set is used.